core/arch/arm/kernel/kern.ld.S

```diff
@@ -58,18 +58,27 @@
 #define ROUNDDOWN(x, y)		(((x) / (y)) * (y))
 #endif
 
+/*
+ * TEE_RAM_VA_START:            The start virtual address of the TEE RAM
+ * TEE_TEXT_VA_START:           The start virtual address of the OP-TEE text
+ */
+#define TEE_RAM_VA_START                CFG_TEE_RAM_START
+#define TEE_TEXT_VA_START               (TEE_RAM_VA_START + \
+			(CFG_TEE_LOAD_ADDR - CFG_TEE_RAM_START))
+
 OUTPUT_FORMAT(CFG_KERN_LINKER_FORMAT)
 OUTPUT_ARCH(CFG_KERN_LINKER_ARCH)
 
 ENTRY(_start)
 SECTIONS
 {
-	. = CFG_TEE_LOAD_ADDR;
-
+	. = TEE_TEXT_VA_START;
+	ASSERT(!(TEE_TEXT_VA_START & (0x20 - 1)),
+		"text start should align to 0x20")
```

Comment:
This alignment will not suite AArch64 that needs 128byte alignment of the vector table.